### PR TITLE
[F] CANDLEPIN-684: Revoke anonymous certificate

### PR DESCRIPTION
**Pull request by JoshAlbrecht23**

### CODE REVIEW

**Comment by nikosmoum** [comment]
I think you may need to backport the GH docker-compose --build changes in this feature branch

**Review by nikosmoum — CHANGES_REQUESTED** [review]

**Comment by nikosmoum** [comment]
> I think you may need to backport the GH docker-compose --build changes in this feature branch

Nevermind, seems there was an intermittent failure from Quay.io, which is now fixed.

File: src/main/java/org/candlepin/resource/ConsumerResource.java [hunk 1]
```diff
@@ -949,8 +957,17 @@ public ConsumerDTO createConsumer(ConsumerDTO dto, String userName, String owner
...
-            // TODO: Revoke anonymous cert
-            anonymousConsumerCurator.delete(anonymPrincipal.getAnonymousCloudConsumer());
+            AnonymousCloudConsumer anonCloudConsumer = anonymPrincipal.getAnonymousCloudConsumer();
+            AnonymousContentAccessCertificate anonCert = anonCloudConsumer.getContentAccessCert();
+            if (anonCert != null) {
+                if (anonCert.getSerial() != null) {
+                    serialCurator.revokeById(anonCert.getSerial().getId());
+                }
+
+                anonymousCertCurator.delete(anonCert);
```
[the remark this answers is not in the document]
Comment (JoshAlbrecht23) [reply src/main/java/org/candlepin/resource/ConsumerResource.java]: Perfect!! That is much better! Updated

**Comment by JoshAlbrecht23** [comment]
> > I think you may need to backport the GH docker-compose --build changes in this feature branch
> 
> Nevermind, seems there was an intermittent failure from Quay.io, which is now fixed.

Yeah, there were issues with Quay.io yesterday and we weren't able to pull images unfortunately

File: src/main/java/org/candlepin/resource/ConsumerResource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikosmoum) [reply src/main/java/org/candlepin/resource/ConsumerResource.java]: You can also remove the serialcurator as an argument from this class (and also from tests)

**Review by nikosmoum — APPROVED** [review]

**Pull request merged**